### PR TITLE
tools/pipeline: add interrupt signal handler

### PR DESCRIPTION
**Pull request by justinjoy**

### CODE REVIEW

**Comment by justinjoy** [comment]
hmm, I don't have a better idea to unref pipeline when user press ctrl+c.

File: tools/pipeline.c [hunk 1]
```diff
@@ -8,21 +8,53 @@
...
+
+  gaeguli_pipeline_remove_target (pipeline, target_id, &error);
+
+  g_debug ("target removed");
+
+  return G_SOURCE_REMOVE;
+}
+
+static void
+stream_stopped_cb (guint target_id, gpointer user_data)
```
Comment (xhaakon) [inline tools/pipeline.c]: Wrong signal handler signature; you're missing `pipeline` argument.
```
stream_stopped_cb (GaeguliPipeline *pipeline, guint target_id, gpointer user_data)
```

File: tools/pipeline.c [hunk 2]
```diff
@@ -8,21 +8,53 @@
...
+
+  fifo = g_object_get_data (G_OBJECT (app), "fifo");
+  target_id = gaeguli_pipeline_add_fifo_target (pipeline, fifo, &error);
+}
+
+static void
+shutdown (GApplication * app, gpointer user_data)
+{
+  g_debug ("app released");
+  g_application_release (app);
```
Comment (xhaakon) [inline tools/pipeline.c]: By the time `GApplication` fires its `"shutdown"` signal, it has been already released. I think you want to move `g_application_release (app);` to `stream_stopped_cb()`.

With that, this callback is unnecessary unless you want to keep the debug message.

File: tools/pipeline.c [hunk 3]
```diff
@@ -8,21 +8,53 @@
...
+
+  return G_SOURCE_REMOVE;
+}
+
+static void
+stream_stopped_cb (guint target_id, gpointer user_data)
+{
+  GApplication *app = user_data;
+
+  g_debug ("stream stopped");
```
Comment (xhaakon) [inline tools/pipeline.c:51]: Put `g_application_release (app);` after this line.

**Review by xhaakon — CHANGES_REQUESTED** [review]

**Pull request merged**